src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
+type FigureColor = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: FigureShape;
+  color: FigureColor;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: FigureShape = 'triangle';
 
-}
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const isValidSides = this.a > 0 && this.b > 0 && this.c > 0;
+
+    if (!isValidSides) {
+      throw new Error('sides cannot be negtive');
+    }
 
-export class Circle {
+    const isSideBiggerThanTwoOthers = (this.a >= (this.b + this.c))
+      || (this.b >= (this.a + this.c))
+      || (this.c >= (this.a + this.b));
 
+    if (isSideBiggerThanTwoOthers) {
+      throw new Error('side cannot be larger than two other sides');
+    }
+  }
```

Comment:
would be nice to move this logic to some private method and use it in the constructor